frontend/src/metabase-types/entities/search.ts

```diff
@@ -0,0 +1,12 @@
+import type {
+  SearchResultId,
+  SearchModel,
+  SearchResult,
+} from "metabase-types/api";
+
+/** SearchResult with most properties optional */
+export type WeakSearchResult<
+  Id extends SearchResultId = SearchResultId,
+  Model extends SearchModel = SearchModel,
+> = Partial<SearchResult<Id, Model>> &
+  Pick<SearchResult<Id, Model>, "id" | "name" | "model">;
```

Comment:
I think this should go closer to the components that use it. It might be generally useful, but it's more likely going to be used once, and will be easier to find closer to where it is actually needed

Reply:
I originally put it in the `browse` folder, but I noticed that it needs to be used in `frontend/test/__support__/server-mocks/search.ts`. I didn't want to import something from `browse` in that file. So I moved it here